src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
```

Comment:
When converting `YY` to `YYYY`, it is assumed that `year` is a string, but since it comes from `split`, it is indeed a string. However, the comparison `year < 30` is using a string in a numerical comparison, which can lead to unexpected results. It's better to convert `year` to a number before doing the comparison.

Reply:
Alright, I'll double-check to be sure.